app/src/main/java/com/humara/nagar/ui/report/ReportFragment.kt

```diff
@@ -23,23 +43,216 @@ class ReportFragment : BaseFragment() {
+            toolBarTitle.text = resources.getString(R.string.reportIssueTitle)
+            toolBarBackButton = root.findViewById(R.id.leftIcon)
+            toolBarHistoryButton = root.findViewById(R.id.rightIcon)
+
+            //Settings up list for spinners
+            inputCategory.setOptions(R.array.category_list)
+            inputLocality.setOptions(R.array.locality_list)
+            inputLocality.setRequiredInput(true)
+            inputComment.apply {
+                switchToMultiLined()
+            }
+
+            inputCategory.setUserInputListener {
+                reportViewModel.setCategory(it)
+                Logger.debugLog(it)
+            }
+            inputLocality.setUserInputListener {
+                reportViewModel.setLocality(it)
+                Logger.debugLog(it)
+            }
+            inputLocation.setUserInputListener {
+                reportViewModel.setLocation(it)
+                Logger.debugLog(it)
+            }
+            inputComment.setUserInputListener {
+                reportViewModel.setComment(it)
+                Logger.debugLog(it)
+            }
+        }
+    }
+
+    private fun getAddress() {
+        val addresses = Utils.getAddressFromLongAndLat(requireContext())
+        val address = addresses[0]
+        val addressLine = address.getAddressLine(0)
+        val city = address.locality
+        val state = address.adminArea
+        val country = address.countryName
+        val postalCode = address.postalCode
+        Logger.debugLog("Address:\nAddressLine-> $addressLine\nCity-> $city\nState-> $state\nCountry-> $country\nPostalCode-> $postalCode")
+        binding.inputLocation.setInput(addressLine)
+        reportViewModel.setLocation(addressLine)
+    }
+
+    private fun checkForLocationPermission() {
+        val permissions = arrayOf(
+            Manifest.permission.ACCESS_FINE_LOCATION,
+            Manifest.permission.ACCESS_COARSE_LOCATION
+        )
+        if (!PermissionsUtils.checkPermissions(requireContext(), permissions)) {
+            PermissionsUtils.requestPermissions(this@ReportFragment, permissions)
+            requestedPermissions.clear()
+            requestedPermissions = permissions.toMutableList()
+        } else getAddress()
+    }
+
+    private fun showPictureDialog() {
+        val pictureDialog = AlertDialog.Builder(requireContext())
+        pictureDialog.setTitle(resources.getString(R.string.selectAction))
+        val pictureDialogItems = arrayOf(
+            resources.getString(R.string.selectFromGallery),
+            resources.getString(R.string.captureFromCamera)
+        )
+        pictureDialog.setItems(
+            pictureDialogItems
+        ) { _, which ->
+            when (which) {
+                0 -> choosePhotoFromGallery()
+                1 -> takePhotoFromCamera()
+            }
+        }
+        val dialog = pictureDialog.create()
+        dialog.show()
+    }
+
+    private fun choosePhotoFromGallery() {
```

Comment:
Define various permissions in PermissionUtil class only. choosePhotoFromGallery() can be extracted in Utils class

Reply:
Fixed.